### PR TITLE
signup test is added

### PR DESCRIPTION
**Pull request by jxczz2**
## Summary

Signup test is added.

## Known Issues

no problem.

## Trello Card

https://trello.com/c/kFXBAXTK

### CODE REVIEW

File: test/user.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        .post('/users/')
+        .send({
+          name: '',
+          lastName: 'torres',
+          email: '',
+          password: ''
+        })
+        .catch(err => {
+          err.should.have.status(400);
+          err.response.should.be.json;
```
Comment (picollomartin) [inline test/user.js]: Use should or expect not both, check this in all file

File: test/user.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        .send({
+          name: 'sarahi',
+          lastName: 'torres',
+          email: 'sarahi12hg@wolox.com',
+          password: 'woloxwoloA1520'
+        })
+        .then(() => {
+          return chai
+            .request(server)
+            .post('/users/')
```
Comment (picollomartin) [inline test/user.js]: Remove trailing `/`, make a generic method for create users

File: test/user.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+          const users = await User.find({
+            where: {
+              name: 'sarahi',
+              lastName: 'torres',
+              email: 'sarahi12hg@wolox.com'
+            }
+          });
+
+          expect(users).to.be.a('object');
+          expect(users.password).to.not.equal('woloxwolox1520');
```
Comment (picollomartin) [inline test/user.js]: What are you checking here ?

File: test/user.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+          res.should.have.status(201);
+          const users = await User.find({
+            where: {
+              name: 'sarahi',
+              lastName: 'torres',
+              email: 'sarahi12hg@wolox.com'
+            }
+          });
+
+          expect(users).to.be.a('object');
```
Comment (picollomartin) [inline test/user.js]: This seems unnecessary

**Review by picollomartin — CHANGES_REQUESTED** [review]

File: test/user.js (hunk 3 above)
Comment (jxczz2) [reply test/user.js]: I check that the password is saved encrypted,should I remove it?

Comment (picollomartin) [reply test/user.js]: That's ok, but check with `woloxwoloA1520` (original password)

File: test/user.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
+const chai = require('chai'),
+  dictum = require('dictum.js'),
+  server = require('./../app'),
+  User = require('../app/models').User,
+  { createUser, userOne, anotherUser } = require('./utilUser'),
+  expect = chai.expect,
+  should = chai.should();
```
Comment (picollomartin) [inline test/user.js]: Remove unused

File: test/user.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+          expect(err.response.body).have.property('message');
+          expect(err.response.body).have.property('internal_code');
+          expect(err.response.body.internal_code).to.equal('saving_error');
+          done();
+        });
+    });
+    it('should fail, email already in use for another user', done => {
+      createUser(userOne)
+        .then(() => {
+          return createUser(userOne);
```
Comment (picollomartin) [inline test/user.js]: Remove brackets and return

File: test/user.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+          expect(err.response.body).have.property('message');
+          expect(err.response.body).have.property('internal_code');
+          expect(err.response.body.internal_code).to.equal('saving_error');
+          done();
+        });
+    });
+    it('should fail creation of   user because some field is empty', done => {
+      chai
+        .request(server)
+        .post('/users/')
```
Comment (picollomartin) [inline test/user.js]: Remove trailing `/`

File: test/user.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+      createUser(anotherUser).catch(err => {
+        expect(err).have.status(400);
+        expect(err.response).be.json;
+        expect(err.response.body).have.property('message');
+        expect(err.response.body).have.property('internal_code');
+        expect(err.response.body.internal_code).to.equal('saving_error');
+        done();
+      });
+    });
+    it('should fail creation of   user because password is no correct', done => {
```
Comment (picollomartin) [inline test/user.js]: Remove spaces

File: test/user.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      chai
+        .request(server)
+        .post('/users/')
```
Comment (picollomartin) [inline test/user.js]: Use function created in utils

File: test/utilUser.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+'use strict';
```
Comment (picollomartin) [inline test/utilUser.js]: Maybe a folder `util/users.js`, for avoid confusing this with a real test

File: test/utilUser.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  name: 'sarahi',
+  lastName: 'torres',
+  email: 'sarahidamaris12hg@woloxi.com',
+  password: 'woloxwoloA1520'
+};
+
+exports.createUser = user =>
+  chai
+    .request(server)
+    .post('/users/')
```
Comment (picollomartin) [inline test/utilUser.js]: Remove trailing `/`

**Review by picollomartin — CHANGES_REQUESTED** [review]
Little changes

File: test/user.js (hunk 7 above)
Comment (jxczz2) [reply test/user.js]: both ? , or just the end?

File: test/user.js (hunk 8 above)
Comment (jxczz2) [reply test/user.js]: you mean the space between of and user?

**Review by picollomartin — APPROVED** [review]

**Pull request merged**